### PR TITLE
Introduce blue collab channel

### PR DESCRIPTION
**Pull request by IanMayo**
Supports #2941 

Setup data to test use of collab channel

### CODE REVIEW

**Comment by IanMayo** [comment]
Hello @lilitkarapetyan - I think I'm making good progress here. The messages now expand properly (we had some invalid data in channel definition).

But, when I try to `Claim` a document, I'm pretty sure the database operation happens ok, but the template doesn't become editable.

Aah, in the console, I do see a warning for `duplicate message id`:
image

Maybe we're keeping the old version, but not switching to the new one.  Maybe the error is in there.

Yes! that was the error.  I _think_ collab editing is working in this branch.  @Tristina1788 - you should be able to test the various state changes in the flow-chart in that presentation.

**Comment by Tristina1788** [comment]
@IanMayo  @lilitkarapetyan 

- Text 'Chat' and 'Response' are hard to see. And When message still haven't had response yet, it shows as the white long button. It makes user don't know what it is

image

=> Text 'Chat' and 'Response' should be bold and white color as 'Feedback' text. And when message doesn't have response, we should not show that part.

- With below requirement

image

image



- Message with status 'InProgress' (Edit), user role 'CanSubmitForReview'should has button 'Submit for review', 'Save'

image

- Message with status 'InProgress' (Edit),  user role 'CanApproved' should has button 'Save', 'Submit for review'

image


- Message with status 'InProgress' (Edit),  user role 'CanRelease' should has button 'Save', 'Submit for review', 'Release'
 
image

- Message with status 'InProgress' (Edit),  user role 'CanUnClaim' should has button 'Save', 'Submit for review', 'Release','Unclaim

image

- I think we have 2 status 'Pending' and 'Review' but now I see it has 'Pending Review'

image

image


image

@IanMayo What is the correct one?

- The UI is quite messing and it's hard to follow the action 

https://github.com/serge-web/serge/assets/107697044/e0f411c6-2a48-4edf-9914-ee61180b8d31

- Here 's config for 'Collab Channel' for more details : 

image

**Comment by lilitkarapetyan** [comment]
Hello @IanMayo , When we make changes to the collabChannel, the edit message doesn't appear right away. We have to refresh the page for it to show.  The problem is that we overlook messages with the same IDs in the changes function.
Screenshot (289)

**Comment by IanMayo** [comment]
> Hello @IanMayo , When we make changes to the collabChannel, the edit message doesn't appear right away. We have to refresh the page for it to show. The problem is that we overlook messages with the same IDs in the changes function. !

Great - thanks for spotting that @lilitkarapetyan - that was the next issue for me to work on :-D

I guess it's a similar fix to this one:
https://github.com/serge-web/serge/pull/2942/files#diff-d7caf9817cf78f3a3c1767872426b69f6414014447896ebcc7e0b23150422551R75

Would you mind fixing it please?

**Comment by lilitkarapetyan** [comment]
> > Hello @IanMayo , When we make changes to the collabChannel, the edit message doesn't appear right away. We have to refresh the page for it to show. The problem is that we overlook messages with the same IDs in the changes function. !
> 
> Great - thanks for spotting that @lilitkarapetyan - that was the next issue for me to work on :-D
> 
> I guess it's a similar fix to this one: https://github.com/serge-web/serge/pull/2942/files#diff-d7caf9817cf78f3a3c1767872426b69f6414014447896ebcc7e0b23150422551R75
> 
> Would you mind fixing it please?

I belive I've resolved it using teh changes function - https://github.com/serge-web/serge/pull/2942/commits/e7ff7d2b0574885dbf0449ee6d99895fb701f0bf

**Comment by IanMayo** [comment]
> I belive I've resolved it using teh changes function - [e7ff7d2](https://github.com/serge-web/serge/commit/e7ff7d2b0574885dbf0449ee6d99895fb701f0bf)

Fix working perfectly, though I've changed a variable name

**Comment by Tristina1788** [comment]
Checking with te latest code on branch '2941_send_collab_message'

- 'Assign' button is not stable. I have to try click many time to make it work. 

https://github.com/serge-web/serge/assets/107697044/a24e36f4-659f-427a-91e8-5f45f6de4f94

- No action happens when approve for message which in status 'Pending Review'


https://github.com/serge-web/serge/assets/107697044/905a40be-b9f7-4c5e-beec-1225a98ddae5



- Above issues still happens

https://github.com/serge-web/serge/pull/2942#issuecomment-2111871936

**Comment by Tristina1788** [comment]
@IanMayo Seem this branch is merged by closing ticket #2941 . So how about the issue on this branch? Should I create a ticket for it?

**Comment by IanMayo** [comment]
> @IanMayo Seem this branch is merged by closing ticket #2941 . So how about the issue on this branch? Should I create a ticket for it?

Yes please :-). I did an early merge so that the collab channel became available in `serge-inet`
